## DV_RProject3
========================================================
#### Group member: Anna Mengjie Yu (my3852),  Duy Vu (dhv242),  Syed Naqvi (san724)

With the advent of next generation sequencing, the sequencing cost per genome is decreasing significantly. Gene content and gene order has been useful in studying species evolution and ecological adaptation. Good data visualization tools will help community better understand the evolution story underneath the data. In this project, we will be using diatom plastid genome data from several diatom species. Each species may or may not share the same gene content due to environment selection. Difference tables can be joined by the gene names.

The tables contain plastid genome gene content, gene length and gene order, and they are accessible from Supporting Information in Yu's publicaiton. [http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0107854]

*********
Load packages

```{r}
source("../01 Data/loadPackage.R", echo = TRUE)
```

*********
First, we displayed a few lines of the dataset stored in oracle database. The table includes Name(gene name), MINIMUM (gene start position), MAXIMUM(gene end position), LENGTH(gene length), DIRECTION(sense and anti-sense strand).

**Table 1: Cyclotella**
```{r}
source("../01 Data/cyclotella.R", echo = TRUE)
```

*********
**Table 2: Thalassiosira weissflogii**
Thalassiosira weissflogii is a closely related species to Cyclotella. They both belong to the same Thalassiosirales family.

```{r}
source("../01 Data/weiss.R", echo = TRUE)
```

*********
**Table 3: Cerataulina daemon**
Cerataulina daemon is more closely related to Chaetoros simplex than the two Thalassiosirales.
```{r}
source("../01 Data/cerataulina.R", echo = TRUE)
```

*********
**Table 4: Chaetoceros simplex**
```{r}
source("../01 Data/chaetoceros.R", echo = TRUE)
```

*********

**Table 5: Thalassiosira oceanica**
```{r}
source("../01 Data/ocean.R", echo = TRUE)
```

*********
 Data Wrangling 1
 
 1.For each original table, create new column gene_family by extracting the first three charaters from gene NAME, and further group by gene family.
```{r}
source("../02 Data Wrangling/geneFamily.R", echo = TRUE)
```
 *********
 Data Wrangling 2
 
 
```{r}
source("../02 Data Wrangling/cyc_FR.R", echo = TRUE)
```

*********
 Data Wrangling 3
 
blalalala
```{r}
#source("../02 Data Wrangling/geneFamily.R", echo = TRUE)
```
 
 *********
 **Figure 1: Comparison of number of genes in different species**
```{r}
source("../02 Data Wrangling/geneFamily.R", echo = TRUE)
source("../03 Visualizations/R3_figure1.R", echo = TRUE)
```
In this figure, the x axis indicates gene family, the y axis indicates gene number in that family. The red color indicates genes in forward direction (sense strand), and the blue color indicates genes in reverse direction (anti-sense strand).

From this figure, we can see that Cyclotella and Thalassiosira weissflogii has similar number of forward and reverse genes in each gene family, while Cerataulina and Chaetoceros showing different number of forward and reverse genes in each gene family. 

*********
**Figure 2: Comparison of number of genes in each gene family in forward and reverse direction**
The number of genes in each gene family, using Cyclotella as an example.
```{r}
source("../03 Visualizations/cyc_total_pie.R", echo = TRUE)
```

The number of forward genes in each gene family, and the number of reverse genes in each gene family.

```{r}
source("../03 Visualizations/pie_fr.R", echo = TRUE)
```

*********
**Figure 3:fdafdafd**

*********
**Figure 4: Histogram from non-categorical data**
```{r}
source("../01 Data/categorical.R", echo = TRUE)
source("../03 Visualizations/histogram.R", echo = TRUE)
```

